src/js/player.js

```diff
@@ -967,15 +977,22 @@ class Player extends Component {
     TECH_EVENTS_RETRIGGER.forEach((event) => {
       this.on(this.tech_, event, this[`handleTech${toTitleCase(event)}_`]);
     });
+
+    Object.keys(TECH_EVENTS_QUEUE).forEach((event) => {
+      this.on(this.tech_, event, () => {
+        if (this.tech_.playbackRate() === 0) {
```

Comment:
I think this would be a breaking change. Potentially, doing it when `isSeeking()` is true *and* `playbackRate` is `0`, might be OK.

Reply:
Could you elaborate what would/could be breaking? A tech or source handler would have to both set the playback rate to zero and rely on these events being dispatched by the _player_ (the tech itself will still dispatch them) right after the seek. Perhaps there's some use case I'm not thinking of.

I don't think that would work. Correct me if I'm wrong but I think `seeking()` is only true between the `seeking` and `seeked` events, and this needs to queue everything after and including the `seeked` event.

Reply:
Because people could still expect these events even when `playbackRate` is `0`.
Maybe we need to allow techs/source handlers dictate when seeking is happening? Looks like `seeking()` method just asks the video element directly:
https://github.com/videojs/video.js/blob/d7f45bab89790fe84f8fcc162df2b9e2ff6d265b/src/js/tech/html5.js#L1552-L1564
https://github.com/videojs/video.js/blob/d7f45bab89790fe84f8fcc162df2b9e2ff6d265b/src/js/tech/html5.js#L1700-L1704

but we could modify it so that sourcehandlers can handle it as well like `duration` and `seekable`? https://github.com/videojs/video.js/blob/master/src/js/tech/tech.js#L1166-L1173

Reply:
> Looks like seeking() method just asks the video element directly

Yeah, that's why I said it wouldn't work. I like the idea of deferring `seeking()` to the source handlers. In fact, that should probably be done anyway to make the API correct -- i.e. you'd expect `seeking()` to be true up to the point a `seeked` is fired but that's not the case with the current code. I'll make that change and gate the event queueing to it.

Any other APIs you can think of that might need to be deferred to spoof things properly?

Reply:
Not sure, but we've been deferring them as needed, we just haven't really needed any others until now, apparently.
Do you think that deferring to SH and then only queueing the events if `seeking()` works?

Reply:
Yeah, that should work.

If you want to go with override-only behaviour, there's potentially another possibility. We could rejigger player.js so that most (all?) of the event handlers are dispatched on the player event rather than the tech event. And then have something like `tech.delegateEvent(event)`. An undelegated event would then just be retriggered on the player, whereas a delegated event would be triggered on a source handler -- and expected to eventually be fired on the player manually be the source handler. I did a quick test and it worked as expected, but you obviously have a better idea about the bigger implications of such a change.

Reply:
I think that deferring `seeking()` makes sense in this case.

I'm not sure I follow this 100%. If I understand correctly, the player would not listen for tech events but rather the tech would trigger the events directly on the player when necessary? If that is the case, I see two issues.
1. the tech and SH aren't supposed to have a reference to the player, so, they can't trigger an event on it.
2. triggering the event on the player directly would mean that the player doesn't necessarily have a chance to do something related to the event before others would receive the event, plus, if the player would to re-trigger the event, it would end up in an event loop.

This also does lend itself a bit to what middleware strives to be and we've been thinking/wanting to have events be able to be intercepted by mw as well, so, maybe we can do that eventually. I think that deferring `seeking()` is the quick solution and a decent solution to this.

Reply:
Yeah, I was talking about removing `this.trigger` to avoid the event loop, but you got the gist of what I was going for.

Reply:
Cool. Yeah, I think that's a bit more complicated and would require a lot of though to get it right.